client/src/context/post/PostReducer.js

```diff
@@ -14,10 +16,25 @@ export default (state, action) => {
         posts: action.payload,
         isLoading: false,
       }
+    case SUBMIT_POST:
+      return {
+        ...state,
+        posts: [{ ...action.payload, comments: [] }, ...state.posts],
+        isLoading: false,
+      }
+    case SUBMIT_COMMENT:
+      return {
+        ...state,
+        posts: state.posts.map((post) =>
```

Comment:
why are you doing what you're doing here? I'm kind of confused, some code comments would be nice

Reply:
adding the new comment / post to the array in state, instead of requerying all of the post/comments or making another api call.